Yoast/yoast-components
# 653 654 collapsible header icon keyword

## Summary

This PR can be summarized in the following changelog entry:

* Adds a subtitle to the collapsible header component

## Relevant technical choices:

- there was the need to make changes to the `Collapsible` component, consequently the `ContentAnalysis` needed some changes too
- the new sub title is passed via the `subTitle` prop
- removes `AnalysisCollapsible` (this should go in the changelog) and use `Collapsible` directly: `AnalysisCollapsible` was duplicating 95% of the code from `Collapsible`
- adds new props to control the SVG icons, including their `ViewBox`: this is necessary because some icons have been coded with a different ViewBox size, thus affecting the way they're rendered
- makes it possible to control the left and right icons independently
- refactors `ContentAnalysis` and uses a `renderCollapsible()` helper function to simplify code
- various CSS adjustments

Ideally, the `Collapsible` header component should be as style-agnostic as possible. For the header itself, it's difficult to style it independently other than passing props. Instead, the content of the collapsible should be styled independently using a specific styled component.

Partially fixes also #671 which will need some adjustments in the plugin. As requested, the left and right paddings should be set on the component itself. This means the StyledSection used in the plugin should be adjusted consequently. Screenshot of the analysis in the plugin (notice the left and right paddings to reset):

<img width="742" alt="screen shot 2018-07-29 at 17 21 39" src="https://user-images.githubusercontent.com/1682452/43382318-c45fecda-93d7-11e8-9c2b-da042d2c1e12.png">


## Test instructions

- check the standalone app (yarn start and go to localhost:3333) and then go in the "Collapsibles" tab
- verify all the collapsible examples there work as expected
- note: in this page I've added a few styles to emulate the WordPress font metrics
- note about the focus style: difficult because there's the need to override some styles that come from the button; I've made it in a way that there's also a subtile feedback when clicking the collapsible
- note about the arrow icons on the right: difficult to align them vertically when there's a subtitle, ideally they should be shifted a few pixels to the bottom
- go in the Content Analysis tab
- verify everything works correctly
- the Content Analysis collapsibles already implement some of the CSS adjustments requested in #671

Fixes #653 
Fixes #654
